ci-operator/config/openshift/os/openshift-os-master.yaml

```diff
@@ -34,6 +38,95 @@ images:
       as:
       - registry.svc.ci.openshift.org/coreos/coreos-assembler:latest
   to: build-image
+- dockerfile_literal: |
+    # Container image used to build RHCOS in Prow build step to be able to push
+    # it and use it as a native container image for other jobs in Prow CI.
+    #
+    # This stage thus builds the RHCOS OCI image just like the test jobs do
+    # before the tests. However, due to limitations within OpenShift CI, we
+    # need to do the following:
+    # 1. Disable KVM by using the COSA_NO_KVM env var. This is due to OpenShift
+    # CI using the default OpenShift Docker Build strategy, which precludes us
+    # from mounting /dev/kvm into the build context.
+    # 2. Due to a limitation in the version of Buildah that OpenShift Builds
+    # uses (at the time of this writing, it uses a v1.23.z version of Buildah),
+    # we cannot mount the build context into the container build (e.g., `RUN
+    # --mount=type=bind,rw=true,src=.,dst=/buildcontext,bind-propagation=shared
+    # <cmd>`), which would allow it to mutate the build context. This is due to
+    # https://github.com/containers/buildah/pull/3548 not being present.
+    # For now, this necessitates passing this image into the cosa-build image
+    # build below to extract the OCI archive and inject it into the
+    # ImageStream. However, once the OpenShift CI system is upgraded to use
+    # OpenShift 4.11, we can create the RHCOS image in a single shot via this
+    # stage.
+    FROM build-image:latest as base
+    ENV COSA_NO_KVM=1
+    # 1. Run only the ostree build step
+    # 2. Clean-up the cache
+    RUN /src/ci/prow-entrypoint.sh build && \
+        rm -rf "${COSA_DIR}/cache"
```

Comment:
Optional: This image build and the below `oci-archive` image build can be combined into a single image build stage like I did in https://github.com/openshift/release/pull/30134.